File: Tareas/tarea11.rb
puts "Serie B"
puts "Ingrese el número de la serie al que se quiere llegar: "
n = gets().to_i
control = 2
contador = 1

for i in(1..n)
  if(i % 2 == 0)then
    for j in (1..control)
      if(contador > n)then
        abort()
      else
        print "#{i} "
        contador+=1
      end
    end
    control+=1
  else
    if(contador>n)then
      abort()
    else
  print "-#{i} "
  contador+=1
    end
  end
end 

File: Tareas/tarea11.java
// Nombre: Miguel A. Pinal Rodriguez     Mat: 1727055    Hora:V6
// Tarea 11 - Serie B
import javax.swing.JOptionPane;

public class T11SB
{
	public static void main (String[] arg)
	{
		String mensaje="";
		int pos=0, i=1, j=2, posdes=0, k=1;
		String Ventana = JOptionPane.showInputDialog("BIENVENIDO A LA SERIE B\n:"
				+"Ingrese hasta que posicion desea llegar en la serie: ");
	
		posdes = Integer.parseInt(Ventana);
		for(pos=1;pos<=posdes;pos++)
		{
			if(i%2==0)
			{
				mensaje=mensaje+i+", ";
				if(k==j)
				{
					i++;
					j++;
					k=1;
				}else {
					k++;
				}
			}else {
				mensaje=mensaje+"-"+i+", ";
				i++;
			}
			
		}
		JOptionPane.showMessageDialog(null,mensaje);
		 
	}
}

